scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/descriptor/JobDescriptorImpl.java

```diff
@@ -568,21 +573,45 @@ public void failed() {
 
     public void pause(TaskId taskId) {
         if (getInternal().getType() == JobType.TASKSFLOW) {
-            TaskDescriptor lt = eligibleTasks.get(taskId);
+            EligibleTaskDescriptor eligibleTaskDescriptor = eligibleTasks.remove(taskId);
 
-            if (lt != null) {
-                pausedTasks.put(taskId, eligibleTasks.remove(taskId));
+            if (eligibleTaskDescriptor != null) {
+                pausedTasks.put(taskId, eligibleTaskDescriptor);
+            }
+        }
+    }
+
+    public void pausedTaskOnError(TaskId taskId) {
+        if (getInternal().getType() == JobType.TASKSFLOW) {
+            TaskDescriptor taskDescriptor = runningTasks.remove(taskId);
+
+            if (taskDescriptor != null) {
+                pausedTasks.put(taskId, (EligibleTaskDescriptor) taskDescriptor);
             }
         }
     }
 
     public void unpause(TaskId taskId) {
         if (getInternal().getType() == JobType.TASKSFLOW) {
-            EligibleTaskDescriptor lt = pausedTasks.get(taskId);
+            EligibleTaskDescriptor eligibleTaskDescriptor = pausedTasks.remove(taskId);
 
-            if (lt != null) {
-                eligibleTasks.put(taskId, lt);
-                pausedTasks.remove(taskId);
+            if (eligibleTaskDescriptor != null) {
+                eligibleTasks.put(taskId, eligibleTaskDescriptor);
+            }
+        }
+    }
+
+    public void restoreInErrorTasks() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)